File: docs/sphinx/snippets/cpp/using/examples/ptsbe/bell_circuit_under_depol_noise.cpp
/*******************************************************************************
 * Copyright (c) 2026 NVIDIA Corporation & Affiliates.                         *
 * All rights reserved.                                                        *
 *                                                                             *
 * This source code and the accompanying materials are made available under    *
 * the terms of the Apache License 2.0 which accompanies this distribution.    *
 ******************************************************************************/

// [Begin PTSBE_Bell]
#include "cudaq/ptsbe/PTSBESample.h"
#include "utils.h"

int main() {
  cudaq::ptsbe::sample_options opts;
  opts.shots = 10000;
  opts.noise = bell_noise_model();

  auto result = cudaq::ptsbe::sample(opts, bell);
  result.dump();
}
// [End PTSBE_Bell]

File: docs/sphinx/snippets/python/using/examples/ptsbe/bell_circuit_under_depol_noise.py
# ============================================================================ #
# Copyright (c) 2026 NVIDIA Corporation & Affiliates.                          #
# All rights reserved.                                                         #
#                                                                              #
# This source code and the accompanying materials are made available under     #
# the terms of the Apache License 2.0 which accompanies this distribution.     #
# ============================================================================ #

#[Begin PTSBE_Bell]
import cudaq
from cudaq import ptsbe
from utils import bell, noise

result = ptsbe.sample(bell, shots_count=10_000, noise_model=noise)
print(result)
#[End PTSBE_Bell]